bftengine/src/bftengine/messages/InternalMessage.hpp

```diff
@@ -63,6 +64,10 @@ using InternalMessage = std::variant<FullCommitProofMsg*,
                                      // Carries PrePrepare message after validation.
                                      PrePrepareCarrierInternalMsg,
 
+                                     // Add Carriers of messages which will translate
```

Comment:
It's not a translation - in fact, CarrierMesssage encapsulates an external message, right?

Reply:
Thanks @yuliasherman , encapsulate is a better word than translate in our scenario, I will update the comment accordingly.
CarrierMessage can be thought of as a generic message which can represent a generic encapsulator message. The encapsulation (or even translation in some case) can happen for many reasons. 
In this PR we can see that the encapsulation is happening to translate an external Message into an internal message using ValidatedMessageCarrierInternalMsg.
ValidatedMessageCarrierInternalMsg<MSG> is giving us a way to do this in a generic manner so that for any new External Message and we don't have to add a new type in InternalMessage variant. All such types are represented by CarrierMessage. 
Tomorrow let's assume that we want to encapsulate for a different reason. (assume hypothetically) we want to encapsulate a request message and translate / treat the message as a reply message. We will create a RequestToReplyMsgConv<MSG> : public CarrierMessage. In that case CarrierMessage will represent either validated message or Reply Message based on the context. There can be other scenarios aswell. 
In current context as encapsulation is happening, I will update the comment.